examples/language-modeling/run_prompt_tuning_clm.py

```diff
@@ -0,0 +1,365 @@
+        default="ought/raft", metadata={"help": "The name of the dataset to use (via the datasets library)."}
+    )
+    dataset_config_name: Optional[str] = field(
+        default="twitter_complaints",
+        metadata={"help": "The configuration name of the dataset to use (via the datasets library)."},
+    )
+    max_eval_samples: Optional[int] = field(
+        default=None,
+        metadata={
+            "help": (
+                "For debugging purposes or quicker training, truncate the number of evaluation examples to this "
+                "value if set."
+            )
+        },
+    )
+
+    streaming: bool = field(default=False, metadata={"help": "Enable streaming mode."})
+
+    max_seq_length: Optional[int] = field(
+        default=64,
+        metadata={
+            "help": "The maximum total input sequence length after tokenization. Sequences longer "
+            "than this will be truncated."
+        },
+    )
+
+    def __post_init__(self):
+        if self.streaming:
+            require_version("datasets>=2.0.0", "The streaming feature requires `datasets>=2.0.0`")
+
+        if self.dataset_name is None:
+            raise ValueError("Need either a dataset name or a training/validation file.")
+
+
+def main():
+    parser = HfArgumentParser((ModelArguments, DataTrainingArguments, GaudiTrainingArguments))
+    model_args, data_args, training_args = parser.parse_args_into_dataclasses()
+    logging.basicConfig(
+        format="%(asctime)s - %(levelname)s - %(name)s -   %(message)s",
+        datefmt="%m/%d/%Y %H:%M:%S",
+        handlers=[logging.StreamHandler(sys.stdout)],
+    )
+    logger.setLevel(logging.INFO if is_main_process(training_args.local_rank) else logging.WARN)
+    if is_main_process(training_args.local_rank):
+        transformers.utils.logging.set_verbosity_info()
+        transformers.utils.logging.enable_default_handler()
+        transformers.utils.logging.enable_explicit_format()
+
+    set_seed(training_args.seed)
+    if model_args.peft_type == "prompt_tuning":
+        peft_config = PromptTuningConfig(
+            task_type=TaskType.CAUSAL_LM,
+            prompt_tuning_init=PromptTuningInit.TEXT,
+            num_virtual_tokens=model_args.num_virtual_tokens,
+            prompt_tuning_init_text="Classify if the tweet is a complaint or not:",
+            tokenizer_name_or_path=model_args.model_name_or_path,
+        )
+    elif model_args.peft_type == "p_tuning":
+        peft_config = PromptEncoderConfig(
+            task_type=TaskType.CAUSAL_LM,
+            num_virtual_tokens=model_args.num_virtual_tokens,
+            encoder_hidden_size=model_args.encoder_hidden_size,
+        )
+    elif model_args.peft_type == "prefix_tuning":
+        peft_config = PrefixTuningConfig(
+            task_type=TaskType.CAUSAL_LM,
+            num_virtual_tokens=model_args.num_virtual_tokens,
+        )
+
+    text_column = "Tweet text"
+    label_column = "text_label"
```

Comment:
are these 2 specific to the dataset in-use? if so, can we check dataset name and put others as unimplemented?

Reply:
yes